File: otel-instrumentation/templates/java/OTelConfig.java
package com.las;

import io.opentelemetry.api.OpenTelemetry;
import io.opentelemetry.api.trace.Tracer;
import io.opentelemetry.exporter.otlp.trace.OtlpGrpcSpanExporter;
import io.opentelemetry.sdk.OpenTelemetrySdk;
import io.opentelemetry.sdk.resources.Resource;
import io.opentelemetry.sdk.trace.SdkTracerProvider;
import io.opentelemetry.sdk.trace.export.BatchSpanProcessor;
import io.opentelemetry.semconv.resource.attributes.ResourceAttributes;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import java.util.concurrent.TimeUnit;

/**
 * Configuração do OpenTelemetry para LAS
 * Auto-gerado pelo auto-instrumentador
 */
@Configuration
public class OTelConfig {

    private static final String LAS_ENDPOINT = System.getenv()
        .getOrDefault("LAS_ENDPOINT", "http://api.soservices.com.br:8443");
    private static final String LAS_TOKEN = System.getenv().get("LAS_TOKEN");
    private static final String SERVICE_NAME = System.getenv()
        .getOrDefault("SERVICE_NAME", "java-service");
    private static final String SERVICE_ENV = System.getenv()
        .getOrDefault("SERVICE_ENVIRONMENT", "production");

    @Bean
    public OpenTelemetry openTelemetry() {
        Resource resource = Resource.getDefault()
            .merge(Resource.builder()
                .put(ResourceAttributes.SERVICE_NAME, SERVICE_NAME)
                .put(ResourceAttributes.SERVICE_VERSION, "1.0.0")
                .put(ResourceAttributes.DEPLOYMENT_ENVIRONMENT, SERVICE_ENV)
                .put("las.token", LAS_TOKEN != null ? LAS_TOKEN : "")
                .build());

        // OTLP Span Exporter
        OtlpGrpcSpanExporter.Builder exporterBuilder = OtlpGrpcSpanExporter.builder()
            .setEndpoint(LAS_ENDPOINT)
            .setTimeout(30, TimeUnit.SECONDS);
        if (LAS_TOKEN != null && !LAS_TOKEN.isBlank()) {
            exporterBuilder.addHeader("Authorization", "Bearer " + LAS_TOKEN);
        }
        OtlpGrpcSpanExporter otlpExporter = exporterBuilder.build();

        SdkTracerProvider tracerProvider = SdkTracerProvider.builder()
            .addSpanProcessor(BatchSpanProcessor.builder(otlpExporter).build())
            .setResource(resource)
            .build();

        OpenTelemetrySdk.getObservableMeterProvider();

        return OpenTelemetrySdk.builder()
            .setTracerProvider(tracerProvider)
            .buildAndRegisterGlobal();
    }

    @Bean
    public Tracer tracer(OpenTelemetry openTelemetry) {
        return openTelemetry.getTracer(SERVICE_NAME);
    }
}

File: otel-instrumentation/templates/dotnet/OTelConfig.cs
using OpenTelemetry;
using OpenTelemetry.Exporter;
using OpenTelemetry.Resources;
using OpenTelemetry.Trace;
using OpenTelemetry.Metrics;
using System;
using Microsoft.Extensions.DependencyInjection;

namespace LAS.Instrumentation
{
    /// <summary>
    /// Configuração do OpenTelemetry para LAS
    /// Auto-gerado pelo auto-instrumentador
    /// </summary>
    public static class OTelConfig
    {
        private static readonly string LasEndpoint = 
            Environment.GetEnvironmentVariable("LAS_ENDPOINT") ?? 
            "https://api.soservices.com.br:8443/api/v1/ingest/otel";
        
        private static readonly string LasToken = 
            Environment.GetEnvironmentVariable("LAS_TOKEN");
        
        private static readonly string ServiceName = 
            Environment.GetEnvironmentVariable("SERVICE_NAME") ?? "dotnet-service";
        
        private static readonly string ServiceEnvironment = 
            Environment.GetEnvironmentVariable("SERVICE_ENVIRONMENT") ?? "production";

        public static IServiceCollection AddLasOpenTelemetry(this IServiceCollection services)
        {
            var resource = ResourceBuilder.CreateDefault()
                .AddService(ServiceName, serviceVersion: "1.0.0")
                .AddAttributes(new Dictionary<string, object>
                {
                    { "deployment.environment", ServiceEnvironment },
                    { "las.token", LasToken ?? "" }
                });

            services.AddOpenTelemetry()
                .WithTracing(tracerProvider =>
                {
                    tracerProvider
                        .SetResourceBuilder(resource)
                        .AddAspNetCoreInstrumentation()
                        .AddHttpClientInstrumentation()
                        .AddSqlClientInstrumentation()
                        .AddOtlpExporter(options =>
                        {
                            options.Endpoint = new Uri(LasEndpoint);
                            if (!string.IsNullOrEmpty(LasToken))
                            {
                                options.Headers = $"Authorization: Bearer {LasToken}";
                            }
                        });
                })
                .WithMetrics(metricsProvider =>
                {
                    metricsProvider
                        .SetResourceBuilder(resource)
                        .AddAspNetCoreInstrumentation()
                        .AddHttpClientInstrumentation()
                        .AddOtlpExporter(options =>
                        {
                            options.Endpoint = new Uri(LasEndpoint);
                        });
                });

            Console.WriteLine($"✓ OpenTelemetry inicializado para {ServiceName}");
            Console.WriteLine($"  Endpoint: {LasEndpoint}");

            return services;
        }
    }
}
